From 0a085bff54c3f02c20412c5f5a4060c2acb20d4c Mon Sep 17 00:00:00 2001
From: Judith Mendez <jm@ti.com>
Date: Mon, 18 Nov 2024 10:26:55 -0600
Subject: [PATCH 1/2] Fix Makefile for to build with yocto

Fix Makefile that builds hwspinlocktest out-of-tree-module
for yocto recipie to be able to build the module.

Upstream-Status: Inactive-Upstream [private repo]
Signed-off-by: Judith Mendez <jm@ti.com>
---
 Makefile | 18 ++++++++++--------
 1 file changed, 10 insertions(+), 8 deletions(-)

diff --git a/Makefile b/Makefile
index 23ee629..d8fe76d 100644
--- a/Makefile
+++ b/Makefile
@@ -3,15 +3,17 @@
 # TI OMAP HwSpinlock Unit Test
 #

-obj-m = omap_hwspinlock_test.o
+obj-m := omap_hwspinlock_test.o
+
+SRC := $(shell pwd)

 all:
-ifeq ($(KERNELDIR),)
-	@echo "Error: KERNELDIR not set, exiting..."
-	@echo "Eg: make ARCH=arm CROSS_COMPILE=arm-linux-gnueabihf- KERNELDIR=<linux-kernel rootdir>"
-	@exit 1
-endif
-	make ${MAKE_OPTS} -C $(KERNELDIR) M=$(PWD) modules
+	$(MAKE) -C $(KERNEL_SRC) M=$(SRC)
+
+modules_install:
+	$(MAKE) -C $(KERNEL_SRC) M=$(SRC) modules_install

 clean:
-	$(RM) -r *.o *.ko *.mod* *.dwo .*.dwo .*.cmd *.symvers modules.order
+	rm -f *.o *~ core .depend .*.cmd *.ko *.mod.c
+	rm -f Module.markers Module.symvers modules.order
+	rm -rf .tmp_versions Modules.symvers
--
2.47.0
